sensors/hall_effect_sensor/rpm/hall_effect_sensor.ino

```diff
@@ -0,0 +1,97 @@
+#include "hall_effect_sensor.h"
+
+
+HallEffectSensor::HallEffectSensor(int pin = 2) : Sensor(pin)
+{
+    init_sensor(pin);
+}
+
+void Sensor::init_sensor(const int pin)
+{
+    Serial.begin(9600);
+    attachInterrupt(digitalPinToInterrupt(pin), interruptFunc, RISING);
+    delay(1000); // Gather data to average.
+}
+
+void HallEffectSensor::interruptFunc()
+{
+    PeriodBetweenPulses = micros() - LastTimeWeMeasured;
+    LastTimeWeMeasured = micros();
+
+    if (PulseCounter >= AmountOfReadings) // If counter for amount of readings reach the set limit:
+    {
+        PeriodAverage = PeriodSum / AmountOfReadings;
+        // Reset pulse counter and sum
+        PulseCounter = 1;
+        PeriodSum = PeriodBetweenPulses;
+
+        // Change the amount of readings we use depending on the period
+        // (High RPM -> More readings to average, Low RPM -> Less readings to be responsive)
+        int RemapedAmountOfReadings = map(PeriodBetweenPulses, 40000, 5000, 1, 10);
+        RemapedAmountOfReadings = constrain(RemapedAmountOfReadings, 1, 10);
+        AmountOfReadings = RemapedAmountOfReadings; // Updates amount
+    }
+    else
+    {
+        PulseCounter++;
+        PeriodSum = PeriodSum + PeriodBetweenPulses; // Add the periods so later we can average.
+    }
+}
+
+void HallEffectSensor::loop()
+{
+    LastTimeCycleMeasure = LastTimeWeMeasured;
+    CurrentMicros = micros();
+
+    // CurrentMicros should always be higher than LastTimeWeMeasured, but in rare occasions that's not true, reset LastTimeWeMeasured.
+    if (CurrentMicros < LastTimeCycleMeasure)
+    {
+        LastTimeCycleMeasure = CurrentMicros;
+    }
+
+    // Calculate the frequency:
+    FrequencyRaw = 10000000000 / PeriodAverage; // Frequency * 10^4 (preserve 4 decimals)
```

Comment:
Im not sure where this strange long number came from. Could you clarify why thats there?